cluster.go

```diff
@@ -1741,3 +1741,76 @@ func diffPeers(peers1, peers2 []peer.ID) (added, removed []peer.ID) {
 	}
 	return
 }
+
+// RepoGC performs garbage collection sweep on all peers' IPFS repo.
+func (c *Cluster) RepoGC(ctx context.Context) (*api.GlobalRepoGC, error) {
+	_, span := trace.StartSpan(ctx, "cluster/RepoGC")
+	defer span.End()
+	ctx = trace.NewContext(c.ctx, span)
+
+	members, err := c.consensus.Peers(ctx)
+	if err != nil {
+		logger.Error(err)
+		return nil, err
+	}
+	lenMembers := len(members)
+
+	ctxs, cancels := rpcutil.CtxsWithCancel(ctx, lenMembers)
+	defer rpcutil.MultiCancel(cancels)
+
+	repoGCsResp := make([]*api.RepoGC, lenMembers, lenMembers)
+
+	errs := c.rpcClient.MultiCall(
+		ctxs,
+		members,
+		"Cluster",
+		"RepoGCLocal",
+		struct{}{},
+		rpcutil.CopyRepoGCSliceToIfaces(repoGCsResp),
+	)
+
+	// clubbing repo gc responses of all peers into one
+	globalRepoGC := api.GlobalRepoGC{PeerMap: make(map[string]*api.RepoGC)}
+	for i, resp := range repoGCsResp {
+		e := errs[i]
+
+		if e == nil {
+			globalRepoGC.PeerMap[peer.IDB58Encode(resp.Peer)] = resp
+			continue
+		}
+
+		if rpc.IsAuthorizationError(e) {
+			logger.Debug("rpc auth error:", e)
+			continue
+		}
+
+		logger.Errorf("%s: error in broadcast response from %s: %s ", c.id, members[i], e)
+
+		resp.Error = e.Error()
+		globalRepoGC.PeerMap[peer.IDB58Encode(resp.Peer)] = resp
```

Comment:
I think you can't do this because the resuIting resp will be an empty object. Either RPC has failed to contact that peer, or it did contact it but the method returned error. In any case, the response will not be set at all -at least for remote peers- even if you have returned it in `RepoGCLocal`.

***This is handled in `globalPinInfoCid`*** (third time I am correcting something because it was not done in the same way as that method). You need to set the peer ID and Peername etc manually (you don't have the peername for the remote peer so you can leave it empty or set it to peer ID).

Reply:
didn't consider that `resp` could be nil here.

In `globalPinInfoCid`, what's the reason for not setting authorization errors in PinInfo structs?
https://github.com/ipfs/ipfs-cluster/blob/45e9a313db0d312970bbc847e51824be0dcede22/cluster.go#L1530-L1533

Reply:
Resp is not nil. Resp is simply not set to anything other than the empty object you made the call with.

> In globalPinInfoCid, what's the reason for not setting authorization errors in PinInfo structs?

An auth error is not something out of the ordinary in follower modes or that the user can fix. I chose to ignore them. We can reverse this decision if needed, but for the moment we should keep consistency.